pkg/metrics/metrics.go

```diff
@@ -0,0 +1,131 @@
+package metrics
+
+import (
+	"github.com/heketi/heketi/apps"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"net/http"
+)
+
+type Metrics struct {
+	app apps.Application
+}
+
+const (
+	namespace = "gluster"
+)
+
+var (
+	up = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "", "up"),
+		"Was the last query of heketi successful.",
+		nil, nil,
+	)
```

Comment:
These calls to NewDesc and its arguments seems a bit repetitive. I wonder it it wouldn't be nicer to have something like:
```golang
func promDesc(name, help string,  variableLabels []string) *promethus.Desc {
    return prometheus.NewDesc(....)
}
```

In order to cut down on the boilerplate. As a 2nd consideration, do you think there's any potential of name conflicats if our namespace="gluster" rather than "heketi"? IOW, if someone was also getting metrics from actual gluster nodes as well as Heketi?

Reply:
I'm on board with both.